FE/src/utils/media.ts

```diff
@@ -35,3 +35,48 @@ export const getSupportedMimeTypes = () => {
   ];
   return types.filter((type) => MediaRecorder.isTypeSupported(type));
 };
+
+export const createMicrophoneVolumeMonitor = (
+  stream: MediaStream,
+  volumeCallback: React.Dispatch<React.SetStateAction<number>>
+) => {
+  const audioContext = new AudioContext();
+  const sourceNode = audioContext.createMediaStreamSource(stream);
+  const analyser = audioContext.createAnalyser();
+  analyser.fftSize = 2048;
+
+  sourceNode.connect(analyser);
+  const pcmData = new Float32Array(analyser.fftSize);
+
+  let intervalId: NodeJS.Timer | number;
+
+  const startMonitoring = () => {
+    intervalId = setInterval(() => {
+      analyser.getFloatTimeDomainData(pcmData);
+
+      let sumSquares = 0;
+      for (let i = 0; i < pcmData.length; i++) {
+        sumSquares += pcmData[i] * pcmData[i];
+      }
```

Comment:
[p-4] 선언형 코드로 바꾸는게 더 보기 좋을거 같은데 어떠신가요?

```suggestion
const sumSquares = pcmData.reduce((sum, value) => sum + value * value, 0);

```

Reply:
언제나 환영합니다!